x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/TokenServiceTests.java

```diff
@@ -1097,12 +1193,4 @@ private static DiscoveryNode addAnotherDataNodeWithVersion(
         setState(clusterService, newStateBuilder.build());
         return anotherDataNode;
     }
-
-    private String generateAccessToken(TokenService tokenService, TransportVersion version) throws Exception {
-        String accessTokenString = UUIDs.randomBase64UUID();
```

Comment:
I stared at this wondering if we should still keep tests which are actually testing with UUIDs for BWC?
Reason I'm not sure if it's beneficial here is because we already have `TokenBackwardsCompatibilityIT` tests. WDYT?

Reply:
I think it's still OK and useful to be able to have a `TokenService` instance that can be made to understand only specific token formats and also be able to feed it different ones. We can easily test the format parsing this way. The `TokenBackwardsCompatibilityIT` is very hard to troubleshoot (I had to do it when commiting the mistake from [44a9ad2](https://github.com/elastic/elasticsearch/pull/97395/commits/44a9ad27c93f2e102490fee9122f886f3ce9869d)), although it is indeed more comprehensive because it does not rely on mocking, so it tests everything including that token documents are stored and retrieved in the expected form.

I plan to remove pre-7.2 tokens in a follow-up. It would've been easier to do that before this change, but it would've taken more time to land the new format feature, and I preferred to prioritize the work that worried me more (adding new functionality rather than removing existing one).

Hope the above answers your point?

Reply:
Makes sense. Thanks.
I only have n additional suggestion to add `test-full-bwc` label to this PR. I think it is useful given the changes.

Reply:
Good tip! I'm going to also mark the PR as "ready-for-review" as I'm not sure full-bwc tests trigger for draft PRs.